### PR TITLE
KT-13550  Fix Boolean.TYPE does not compile

### PR DESCRIPTION
**Pull request by takahirom**
#KT-13550 Fixed
https://youtrack.jetbrains.com/issue/KT-13550

### CODE REVIEW

File: j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt [hunk 1]
```diff
@@ -232,29 +232,17 @@ class DefaultExpressionConverter : JavaElementVisitor(), ExpressionConverter {
...
     override fun visitClassObjectAccessExpression(expression: PsiClassObjectAccessExpression) {
         val operand = expression.operand
+        val type = converter.convertTypeElement(operand, Nullability.NotNull)
         val typeName = operand.type.canonicalText
         val primitiveType = JvmPrimitiveType.values().firstOrNull { it.javaKeywordName == typeName }
-        val wrapperTypeName = if (primitiveType != null) {
-            primitiveType.wrapperFqName
+        val name = if (primitiveType != null) {
+            "javaPrimitiveType"
+        } else if (typeName == "void") {
```
Comment (yole) [inline j2k/src/org/jetbrains/kotlin/j2k/ExpressionConverter.kt]: I'd merge these two `if` branches

File: j2k/testData/fileOrElement/classExpression/primitivesAndArrays.kt [hunk 2]
```diff
@@ -1,8 +1,9 @@
 object A {
     @JvmStatic fun main(args: Array<String>) {
-        println(Void.TYPE)
-        println(Integer.TYPE)
-        println(java.lang.Double.TYPE)
+        println(Unit::class.javaPrimitiveType)
```
Comment (yole) [inline j2k/testData/fileOrElement/classExpression/primitivesAndArrays.kt]: This doesn't actually work: `Unit` has no `javaPrimitiveType`. For `void.class`, the previous variant of the code should be used.

Comment (takahirom) [inline j2k/testData/fileOrElement/classExpression/primitivesAndArrays.kt]: Done!

**Comment by yole** [comment]
Thanks a lot for your contribution!

**Pull request merged**